scripts/lib/api/generateApiComponents.ts

```diff
@@ -350,3 +399,44 @@ export async function htmlSignatureToMd(
     .replace(/^`/, "")
     .replace(/`$/, "");
 }
+
+function getHeaderLevel($: CheerioAPI, $main: Cheerio<any>, $dl: Cheerio<any>) {
+  const priorHeaderLevel = getPriorHeaderLevel($, $main, $dl);
+  if (priorHeaderLevel) {
+    if (+priorHeaderLevel == 6) {
+      throw new Error("API component cannot set inexisting header: <h7>");
+    }
+
+    return Math.max(3, +priorHeaderLevel + 1);
+  }
+
+  // Minimum header level for components without a dedicated page. This components should
+  // have an <h1> for the page and <h2> to define the section they belong to.
+  return 3;
+}
+
+function getPriorHeaderLevel(
+  $: CheerioAPI,
+  $main: Cheerio<any>,
+  $dl: Cheerio<any>,
+): string | undefined {
+  const siblings = $dl.siblings();
+  for (const sibling of siblings) {
+    const $sibling = $(sibling);
+    if ($sibling.data("header-type")) {
+      continue;
+    }
+    const tagName = $sibling.get(0)?.tagName;
+    if (tagName?.match(/h[1-6]/)) {
+      return tagName.substring(1);
+    }
+  }
+
+  // If there's no header among the siblings, we look for the prior inline class in some
+  // parent node previously set
```

Comment:
```suggestion
  // If there's no header among the siblings, we look if the component belongs inside a class.
  // Otherwise, we return undefined.
```